File: tests/e2e/scripts/rest_params.ts
function sum(...nums: number[]): number {
    let total: number = 0;
    for (const n of nums) {
        total += n;
    }
    return total;
}

function countArgs(...args: number[]): number {
    return args.length;
}

function main(): void {
    // Basic rest param call
    console.log("sum:", sum(1, 2, 3));

    // No rest args
    console.log("sum_empty:", sum());

    // More args
    console.log("sum5:", sum(1, 2, 3, 4, 5));

    // Count args
    console.log("count:", countArgs(10, 20, 30, 40));

    // Spread call
    const arr: number[] = [10, 20, 30];
    console.log("sum_spread:", sum(...arr));
}

File: tests/e2e/scripts/rest_params.rs
fn sum(nums: Vec<f64>) -> f64 {
    let mut total: f64 = 0.0;
    for n in nums {
        total = total + n;
    }
    total
}

fn countArgs(args: Vec<f64>) -> f64 {
    args.len() as f64
}

fn main() {
    println!("{} {}", "sum:", sum(vec![1.0, 2.0, 3.0]));
    println!("{} {}", "sum_empty:", sum(vec![]));
    println!("{} {}", "sum5:", sum(vec![1.0, 2.0, 3.0, 4.0, 5.0]));
    println!("{} {}", "count:", countArgs(vec![10.0, 20.0, 30.0, 40.0]));
    let mut arr: Vec<f64> = vec![10.0, 20.0, 30.0];
    println!("{} {}", "sum_spread:", sum(arr));
}
